lib/python/picongpu/picmi/interaction/ionization/fieldionization/ADK.py

```diff
@@ -0,0 +1,47 @@
+"""
+This file is part of PIConGPU.
+Copyright 2024 PIConGPU contributors
+Authors: Brian Edward Marre
+License: GPLv3+
+"""
+
+from .fieldionization import FieldIonization
+
+from .....pypicongpu.species.constant.ionizationcurrent import None_
+from .....pypicongpu.species.constant.ionizationmodel import (
+    ADKLinearPolarization,
+    ADKCircularPolarization,
+    IonizationModel,
+)
+
+from ..... import pypicongpu
+
+import enum
+import typeguard
+
+
+@typeguard.typechecked
+class ADKVariant(enum.Enum):
+    LinearPolarization = 0
+    CircularPolarization = 1
+
+
+@typeguard.typechecked
+class ADK(FieldIonization):
+    """Barrier Suppression Ioniztion model"""
+
+    MODEL_NAME: str = "ADK"
+
+    ADK_variant: ADKVariant
+    """extension to the BSI model"""
+
+    def get_as_pypicongpu(self) -> IonizationModel:
+        self.check()
+
+        if self.ADK_variant is ADKVariant.LinearPolarization:
+            return ADKLinearPolarization(ionization_current=None_())
+        if self.ADK_variant is ADKVariant.CircularPolarization:
+            return ADKCircularPolarization(ionization_current=None_())
+
+        # unknown/unsupported ADK variant
+        pypicongpu.util.unsupported(f"ADKVariant {self.ADK_variant}")
```

Comment:
Is a raise or return missing here? 
```suggestion
        raise pypicongpu.util.unsupported(f"ADKVariant {self.ADK_variant}")
```

Reply:
You are right, switched to raising a Value Error since there is now way to recover from an unknown ADK Variant.